devops.sh

```diff
@@ -170,181 +176,202 @@ function toHelmArray() {
-      echo "List of licences we need to check"
-      echo "*******************************************************************"
-      ncl
-      popd
-    done
-    ;;
-
-  "build"|"b")
-    if [ -n "$2" ]; then
       BUILD_TARGET_IMAGE=$BUILD_TARGET
-      echo "############### Build $2 ##########"
-      # if [ "$2" == "converter" ]; then
+      echo "############### Build $image ##########"
+
+      # if [ "$image" == "converter" ]; then
       #   BUILD_TARGET_IMAGE=$CLOUD_TYPE
       # fi
-      # if [ "$2" == "mongodb-backup" ]; then
+      # if [ "$image" == "mongodb-backup" ]; then
       #   BUILD_TARGET_IMAGE=$CLOUD_TYPE
       # fi
-      # if [ "$2" == "activities-importer" ]; then
+      # if [ "$image" == "activities-importer" ]; then
       #   BUILD_TARGET_IMAGE=$CLOUD_TYPE
       # fi
       docker build \
         --target $BUILD_TARGET_IMAGE \
         $(npmBuildSecret) \
-        -t $(imageRef $2) \
-        -f ${images[$2]}/${Dockerfiles[$2]} \
-        ${images[$2]}
-    else
-      for image in "${!images[@]}"; do
-        BUILD_TARGET_IMAGE=$BUILD_TARGET
-        echo "############### Build $image ##########"
-
-        # if [ "$image" == "converter" ]; then
-        #   BUILD_TARGET_IMAGE=$CLOUD_TYPE
-        # fi
-        # if [ "$image" == "mongodb-backup" ]; then
-        #   BUILD_TARGET_IMAGE=$CLOUD_TYPE
-        # fi
-        # if [ "$image" == "activities-importer" ]; then
-        #   BUILD_TARGET_IMAGE=$CLOUD_TYPE
-        # fi
-        docker build \
-          --target $BUILD_TARGET_IMAGE \
-          $(npmBuildSecret) \
-          -t $(imageRef $image) \
-          -f ${images[$image]}/${Dockerfiles[$image]} \
-          ${images[$image]}
-      done
-    fi
-    ;;
-
-  "clean")
-    cleanJobs
-    ;;
-
-  "bush"|"bp")
-    $0 build ${@:2}
-    $0 push ${@:2}
-    ;;
-
-  "deploy")
-    # be sure to create namespace first
-    kubectl --context $KUBE_CONTEXT create namespace $KUBE_NAMESPACE || true
-    copyRegCred
-    helm upgrade \
-      --kube-context $KUBE_CONTEXT \
-      $RELEASE $chartPath \
-      --install \
-      --namespace $KUBE_NAMESPACE \
-      $HELM_DEPLOY_PARAMS \
-      --set image.tag=$(imageTag) \
-      --set image.repository=${IMAGE_REGISTRY}${IMAGE_REGISTRY_PATH:-""} \
-      --set defaultDnsDomain=$DEFAULT_DNS_DOMAIN \
-        ${@:2}
-    ;;
-
-  "destroy")
-    echo "Are you sure you want to destroy the release $RELEASE"
-    select result in Yes No
-    do
+        -t $(imageRef $image) \
+        -f ${images[$image]}/${Dockerfiles[$image]} \
+        ${images[$image]}
+    done
+  fi
+  ;;
+
+"clean")
+  cleanJobs
+  ;;
+
+"bush" | "bp")
+  $0 build ${@:2}
+  $0 push ${@:2}
+  ;;
+
+"deploy")
+  # be sure to create namespace first
+  kubectl --context $KUBE_CONTEXT create namespace $KUBE_NAMESPACE || true
+  copyRegCred
+  DOCKER_IMAGES=""
+  for image in "${!images[@]}"; do
+    imagename=${image//-/}
+    DOCKER_IMAGES+="   --set image.$imagename=$(imageRef $image) "
```

Comment:
I find `image.$imagename` to be less explicit than the old `backend.image.name` but on the other hand I think it's better that devops.sh no longer has to pass through the `image.repository` and the `image.tag`. I pushed a couple of commits that remove the old stuff from values.yaml and deletes L271 as it's no longer needed.